### DISCUSSION TITLE
Inconsistent default geometry column name for postgis

### DISCUSSION THREAD

**Discussion by ondave** [geopandas/geopandas#2710] [Q&A]
It's not a major issue, but I am wondering why the default geometry column name is "geom" for read_postgis? To create a GeoDataFrame, a "geometry" column must be supplied. However the default column name for read_postgis is "geom". So combining a write and read with default arguments fails:

```
import geopandas
from shapely.geometry import Point
from sqlalchemy import create_engine

URI="postgresql:///user:password@localhost:5432/test"
con=create_engine(URI)

d = {'col1': ['name1', 'name2'], 'geometry': [Point(1, 2), Point(2, 1)]}
gdf = geopandas.GeoDataFrame(d, crs="EPSG:4326")

gdf.to_postgis("my_table",con)
gdf_read=geopandas.read_postgis("SELECT * FROM my_table",con)

```
Exception has occurred: ValueError
Query missing geometry column 'geom'

**Comment by martinfleis** [comment]
PostGIS default name for a geometry column is `geom`. Since geopandas uses `geometry` as a default, it causes this inconsistency. 

If you read PostGIS table which is not created by geopandas, the default in most cases works as it is (`geom`), that is why it is there.

We may consider changing default in `to_postgis` which would rename `geometry` to `geom` but I am not entirely sure about that.
